TTD/docs/TTDLiveRecorder.h/ILiveRecorder_DumpHeaps.md

```diff
@@ -0,0 +1,40 @@
+# TTD::ILiveRecorder::DumpHeaps method
+
+Method of the [`TTD::ILiveRecorder`](interface-ILiveRecorder.md) interface.
+
+Adds to the trace file all the data of the process heaps.
+This includes all data allocated with C/C++ functions like `malloc` and `operator new`.
+It also includes data allocated with Windows APIs like
+[`HeapAlloc`](https://learn.microsoft.com/en-us/windows/win32/api/heapapi/nf-heapapi-heapalloc)
+but it excludes data allocated with Windows APIs like
+[`LoadLibrary`](https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibrarya),
+[`VirtualAlloc`](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc) and
+[`MapViewOfFile`](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-mapviewoffile).
+
+```C++
+void TTD::ILiveRecorder::DumpHeaps() noexcept;
+```
+
+## Correct use
+
+This method operates similarly to [`DumpSnapshot`](ILiveRecorder_DumpSnapshot.md) as called with `synchronous` set to `true`.
+It may be freely called from multiple threads simultaneously, it protects itself against possible simultaneous de-commit
+and may record "torn" data if it's simultaneously written. See [`DumpSnapshot`](ILiveRecorder_DumpSnapshot.md) for further details.
+
+This method has no use restrictions.
+
+Note that this method might record a large amount of data in the trace file.
+TTD doesn't perform any differential or incremental snapshotting,
+so the entirety of the process heaps will be recorded each time the mnethod is called.
```

Comment:
feels like we should say what the default behavior for ttd.exe is w.r.t. recording heaps at process startup, otherwise many people who use this API could think "yeah I want to make sure heap memory is available" and bloat trace files unnecessarily.

Reply:
Perhaps more generally, a section that explains how TTD captures memory and an explanation of why you would want/need any of these dump functions.

To the uninitiated, it's not clear why any of these are necessary. Shouldn't TTD just take care of this? ;)

Reply:
Yes, some more general documentation describing the basics of TTD's operation is necessary. Need to explain behaviors such as the full dump (which obviously (:-P)doesn't happen with -recordMode manual, BTW), and also memory/execution model, reasons for data to be incorrect on replay, etc...

That I'm leaving for "later". One step at a time.